### PR TITLE
Task: Add i18n support for public dashboards

### LINKED ISSUE #74347: Task: Public dashboards i18n
Add support for i18n to public dashboards components (config modal, audit table and active users table)

[Start here](https:/github.com/grafana/grafana/blob/main/contribute/internationalization.md)

### PR DESCRIPTION
**Pull request by hainenber**
**What is this feature?**

Add i18n support for `Public dashboards` components

**Why do we need this feature?**

This helps people to use Grafana's `Public dashboards` in their native language.

**Who is this feature for?**

Non-English users

**Which issue(s) does this PR fix?**:

Fixes #74347 

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by lucychen-grafana** [comment]
@hainenber, testing your branch in locally and found an issue with missing text under the acknowledgment section. Can you confirm it is same for you? 
image

**Review by lucychen-grafana — CHANGES_REQUESTED** [review]
Really good job on adding i18n support, left a couple of comments. Also there are a couple of .tsx files under [ConfigPublicDashboard](https://github.com/grafana/grafana/tree/main/public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard) missed that also needs to be marked up for translation as part of this story (config modal). 
image

File: public/app/features/admin/UserListPublicDashboardPage/DashboardsListModalButton.tsx [hunk 1]
```diff
@@ -18,15 +19,26 @@ export const DashboardsListModal = ({ email, onDismiss }: { email: string; onDis
   const { data: dashboards, isLoading } = useGetActiveUserDashboardsQuery(email);
 
   return (
-    <Modal className={styles.modal} isOpen title="Public dashboards" onDismiss={onDismiss}>
+    <Modal
+      className={styles.modal}
+      isOpen
+      title={t('dashboard-list-modal-button.mode.public-dashboard.modal-title', 'Public dashboards')}
```
[the remark this answers is not in the document]
Comment (hainenber) [reply public/app/features/admin/UserListPublicDashboardPage/DashboardsListModalButton.tsx:25]: Thanks! I've fixed in latest commit

[the remark this answers is not in the document]
Comment (hainenber) [reply public/app/features/admin/UserListPublicDashboardPage/DashboardsListModalButton.tsx]: Thanks for the review! I've fixed this in latest commit. PTAL when you have time

**Comment by hainenber** [comment]
> @hainenber, testing your branch in locally and found an issue with missing text under the acknowledgment section. Can you confirm it is same for you?

Yes, I managed to fix this in latest commit. PTAL when you have time!

**Review by lucychen-grafana — COMMENTED** [review]
Can you please also make the changes to [DeletePublicDashboardModal ](https://github.com/grafana/grafana/blob/main/public/app/features/manage-dashboards/components/PublicDashboardListTable/DeletePublicDashboardModal.tsx). Also the [tab title](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/ShareModal/ShareModal.tsx#L61).  Sorry I missed this from previous review

**Comment by lucychen-grafana** [comment]
> > @hainenber, testing your branch in locally and found an issue with missing text under the acknowledgment section. Can you confirm it is same for you?
> 
> Yes, I managed to fix this in latest commit. PTAL when you have time!

I still have the missing text...which commit fixed it?

**Comment by hainenber** [comment]
> Can you please also make the changes to [DeletePublicDashboardModal ](https://github.com/grafana/grafana/blob/main/public/app/features/manage-dashboards/components/PublicDashboardListTable/DeletePublicDashboardModal.tsx). Also the [tab title](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/ShareModal/ShareModal.tsx#L61). Sorry I missed this from previous review

Done! All of them have been marked up with i18n :D. PTAL again

**Comment by hainenber** [comment]
> > > @hainenber, testing your branch in locally and found an issue with missing text under the acknowledgment section. Can you confirm it is same for you?
> > 
> > 
> > Yes, I managed to fix this in latest commit. PTAL when you have time!
> 
> I still have the missing text...which commit fixed it?

My fault for not pushing the changes :(. It's this [commit](https://github.com/grafana/grafana/pull/77674/commits/3a7a839f9bd5bc6d155572e20aecbfeb6a929b77). It's good on my side, pls help checking if it is like that on your side too. Thanks!

**Comment by lucychen-grafana** [comment]
Going to finish the work in https://github.com/grafana/grafana/pull/79659

**Comment by hainenber** [comment]
Go ahead, this has been stale for way too long. I'll close this one off.

**Pull request closed without merging**